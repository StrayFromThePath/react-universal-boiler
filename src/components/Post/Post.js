// @flow
import React from 'react';
import styles from './style.css';

type Props = {
  title: string,
  body: string,
};

const Post = ({ title, body }: Props) => {
  return (
    <div className={styles.pCard}>
      <h2 className={styles.pHeader}>{title}</h2>
      <p className={styles.pBody}>{body}</p>
    </div>
  );
};

export default Post;
