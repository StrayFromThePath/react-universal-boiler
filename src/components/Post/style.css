
.pCard {
  background-color: #fff;
  margin-bottom: 10px;
  box-shadow: 0 4px 5px 0 rgba(0, 0, 0, 0.14), 0 1px 10px 0 rgba(0, 0, 0, 0.12),
    0 2px 4px -1px rgba(0, 0, 0, 0.4);
  margin-left: 10px;
  margin-right: 10px;
  padding: 1em;
}

.pHeader {
  font-size: 1.3em;
  font-weight: 300;
  font-family: 'Rubik';
}

.pBody {
  font-size: 16px;
}
