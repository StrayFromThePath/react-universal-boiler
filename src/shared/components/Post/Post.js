/* @flow */
import React from 'react';
import styled from 'styled-components';

const PostCard = styled.div`
  background-color: #fff;
  margin-bottom: 10px;
  box-shadow: 0 4px 5px 0 rgba(0, 0, 0, .14),
  0 1px 10px 0 rgba(0, 0, 0, .12),
  0 2px 4px -1px rgba(0, 0, 0, .4);
  margin-left: 10px;
  margin-right: 10px;
  padding: 1em;
`;

const Head = styled.h2`
  font-size: 20px;
  font-weight: 300;
`;
const Body = styled.p`
  font-size: 1rem;
`;

type Props = {
  title: string,
  body: string,
};

const Post = ({ title, body }: Props) => {
  return (
    <PostCard>
      <Head>{title}</Head>
      <Body>{body}</Body>
    </PostCard>
  );
};

export default Post;
