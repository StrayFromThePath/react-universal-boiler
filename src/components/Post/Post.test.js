import React from 'react';
import renderer from 'react-test-renderer';
import Post from './Post';

jest.useFakeTimers();
Date.now = jest.fn(() => 1482363367071);

it('renders correctly', () => {
  const tree = renderer.create(<Post title="Foo" body="Bar" />).toJSON();
  expect(tree).toMatchSnapshot();
});
