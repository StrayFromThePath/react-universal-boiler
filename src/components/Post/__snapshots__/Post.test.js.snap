// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="pCard"
>
  <h2
    className="pHeader"
  >
    Foo
  </h2>
  <p
    className="pBody"
  >
    Bar
  </p>
</div>
`;
